### PR TITLE
Add summary sub header to each page

### LINKED ISSUE #19251: [nit] Add summary sub-header to each page of documentation
### Description

Add a “this section will help you:” at the top of each page (similar to Performance Tuning)

### PR DESCRIPTION
**Pull request by triciasfu**
Closes https://github.com/ray-project/ray/issues/19251

### CODE REVIEW

File: doc/source/serve/core-apis.rst [hunk 1]
```diff
@@ -2,6 +2,12 @@
 Core API: Deployments
 =====================
 
+This section should help you:
+
+- create, expose, update and configure deployments
```
Comment (edoakes) [inline doc/source/serve/core-apis.rst:7]: "expose" is weird wording

Comment (edoakes) [reply doc/source/serve/core-apis.rst:7]: maybe "query" ?

File: doc/source/serve/index.rst [hunk 2]
```diff
@@ -4,6 +4,11 @@
 Serve: Scalable and Programmable Serving
 ========================================
 
+This section should help you:
+
+- understand an overview of Ray Serve
+- get started with a Ray Serve example
+
```
Comment (edoakes) [inline doc/source/serve/index.rst:11]: let's not add it to the index IMO

File: doc/source/serve/ml-models.rst [hunk 3]
```diff
@@ -2,6 +2,11 @@
 Serving ML Models
 =================
 
+This section should help you:
+
+- Batch requests to optimize performance
+- Compose separate individual models into a single deployment
```
Comment (edoakes) [inline doc/source/serve/ml-models.rst:8]:
```suggestion
- Serve multiple models by composing deployments
```

File: doc/source/serve/core-apis.rst [hunk 4]
```diff
@@ -2,6 +2,12 @@
...
+- configure resources of your deployments
```
Comment (edoakes) [inline doc/source/serve/core-apis.rst:8]:
```suggestion
- configure resources for your deployments
```

**Review by simon-mo — COMMENTED** [review]
capitalization of first word in each bullet point is inconsistent 😢

File: doc/source/serve/ml-models.rst [hunk 5]
```diff
@@ -2,6 +2,11 @@
 Serving ML Models
 =================
 
+This section should help you:
+
+- Batch requests to optimize performance
+- Serve multiple models by composing deployments
```
Comment (edoakes) [inline doc/source/serve/ml-models.rst:8]: make these lower case to match others

**Review by edoakes — APPROVED** [review]
LGTM, one nit

**Pull request merged**